### ISSUE TITLE
Can't read files containing emoji after interacting with pydavinci & while DaVinci Resolve is open

### ISSUE THREAD

**Issue by seb26** [pedrolabonia/pydavinci#40]
Hi Pedro & @in03,

Bit of a weird one I've come across. Hoping you can help me to identify if this is something in my environment, or an actual bug, and where it might be located.

**Describe the bug**
With a pydavinci Resolve object established and DaVinci Resolve open, reading files via open() or YAML will be done using ASCII decoder. Instead of UTF8 decoder.

Therefore if there is Emojis in the file, UnicodeDecodeError is thrown.

This occurs only when Resolve is open - and therefore the Resolve API is accessible. When Resolve is not open, the file is read correctly as UTF8.

**To Reproduce**

**Python 3.6**:
- Use a Python environment with python 3.6.15, pydavinci 0.2.3 and pyyaml 6.0.1
- Check out https://github.com/seb26/pydavinci_unicode_issue and run `test.py`
- Have DaVinci Resolve 18.6 open

```
% pyenv which python
/Users/data/.pyenv/versions/3.6.15/bin/python
% python test.py                                                   
Begin: ✅ Emoji on stdout
---
Try to load simple file as string
    data.txt: ✅✅✅ data !! ✅✅✅
---
Try to load YAML file with emojis
{'normal_string': 'text here', 'string_with_emojis': '✅✅✅'}
---
Try to import pydavinci
Imported pydavinci.
✅ Emoji after pydavinci import
---
Try to create a Resolve object/contact API
Resolve (0x0x11a90f718) [App: 'Resolve' on 127.0.0.1, UUID: 5bb9cd9e-6912-4337-ab06-280f3effba0f]
✅ Emoji after printing Resolve obj
---
Try to load simple file as string again
Traceback (most recent call last):
  File "test.py", line 31, in <module>
    print('data.txt:', stream.read())
  File "/Users/data/.pyenv/versions/3.6.15/lib/python3.6/encodings/ascii.py", line 26, in decode
    return codecs.ascii_decode(input, self.errors)[0]
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 0: ordinal not in range(128)
```

The result with implementation specific option: `-X utf8`: 
Same UnicodeDecodeError.

**Python 3.10**:
With an environment of Python 3.10, pydavinci@resolve18 0.3.1, and pyyaml 6.0.1.
Also have DaVinci Resolve open.
```
% pyenv which python
/Users/data/.pyenv/versions/3.10.10/bin/python
% python test.py
...
Try to load simple file as string again
Traceback (most recent call last):
  File "test.py", line 31, in <module>
    print('data.txt:', stream.read())
  File "/Users/data/.pyenv/versions/3.6.15/lib/python3.6/encodings/ascii.py", line 26, in decode
    return codecs.ascii_decode(input, self.errors)[0]
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 0: ordinal not in range(128)
```

The result with `-X utf8`:
```
% python -X utf8 test.py
...
Try to load simple file as string again
data.txt: ✅✅✅ data !! ✅✅✅
✅ Emoji after reading back a file
---
Try to load emojis again...
{'normal_string': 'text here', 'string_with_emojis': '✅✅✅'}
✅ Emoji after safe loading yaml file
---
```

**Expected behavior**
Be able to read a UTF-8 file containing emojis and other characters - both via builtin open() and YAML.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Intel-based MacBook Pro 16" - macOS Monterey 12.6.5
 - Python Version - 3.6 and 3.10 as above. Using Pyenv.
 - Davinci Resolve Version - 18.6.4 Build 6

Also reproduced on a Mac Mini (M1, 2020) with macOS Sonoma 14.2, Resolve 18.6.4 and a similar pyenv environment.

**Additional context**
I am writing an app to watch Resolve render job events and send notifications. Notification text is stored in an external user config file in YAML format. Ideally I am able to read text from this file that contains emojis, and use it within the app/send notifs.

**Comment by seb26** [comment]
I think I can conclude that this is not specifically caused by `pydavinci`.

Using `test_defaultResolveAPI.py` in Python 3.11 and Python 3.6.8 (the latest available Python macOS x86_64 binary for 3.6 on python.org). I read text from the emoji file in Step 1 and then imported the Resolve API directly from the fusionscript.so path.

```
% python test_defaultResolveAPI.py 
/Volumes/seb/my/path/to/pydavinci_unicode_issue/test_defaultResolveAPI.py:1: DeprecationWarning: the imp module is deprecated in favour of importlib and slated for removal in Python 3.12; see the module's documentation for alternative uses
  import imp
Python version: 3.10.10 (main, Jan 18 2024, 07:10:51) [Clang 14.0.0 (clang-1400.0.29.202)]
Sys path: ['/Volumes/seb/my/path/to/pydavinci_unicode_issue', '/Users/data/.pyenv/versions/3.10.10/lib/python310.zip', '/Users/data/.pyenv/versions/3.10.10/lib/python3.10', '/Users/data/.pyenv/versions/3.10.10/lib/python3.10/lib-dynload', '/Users/data/.pyenv/versions/3.10.10/lib/python3.10/site-packages']

1. Try to load simple file as string
    data.txt: ✅✅✅ data !! ✅✅✅
---
Import fusionscript.so
Done.
2. Try to load simple file as string AGAIN
    data.txt: ✅✅✅ data !! ✅✅✅
---
Establish a Resolve() object.
Done.
3. Try to load simple file as string AGAIN
Traceback (most recent call last):
  File "/Volumes/seb/my/path/to/pydavinci_unicode_issue/test_defaultResolveAPI.py", line 28, in <module>
    print('    data.txt:', stream.read())
  File "/Users/data/.pyenv/versions/3.10.10/lib/python3.10/encodings/ascii.py", line 26, in decode
    return codecs.ascii_decode(input, self.errors)[0]
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 0: ordinal not in range(128)
```

This result above was the same for 3.10 and 3.6.8.

Now, to test for a result inside DaVinci Resolve console. This is using the 3.6.8 as mentioned before, and printing the sys version inside the Console to affirm that.

```
Python version: 3.6.8 (v3.6.8:3c6b436a57, Dec 24 2018, 02:04:31) 
[GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]
Sys path: ['/Applications/DaVinci Resolve/DaVinci Resolve.app/Contents/MacOS', '/Library/Frameworks/Python.framework/Versions/3.6/lib/python36.zip', '/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6', '/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/lib-dynload', '/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages']
1. Try to load simple file as string
    data.txt: ✅✅✅ data !! ✅✅✅
---
Import fusionscript.so
Done.
2. Try to load simple file as string AGAIN
    data.txt: ✅✅✅ data !! ✅✅✅
---
Establish a Resolve() object.
Done.
3. Try to load simple file as string AGAIN
Traceback (most recent call last):
  File "<nofile>", line 28, in <module>
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/encodings/ascii.py", line 26, in decode
    return codecs.ascii_decode(input, self.errors)[0]
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 0: ordinal not in range(128)

```

You could perhaps to close this issue if it is not `pydavinci` related.

However I would definitely appreciate some eyes on it to see if I can come to a kind of resolution.

**Comment by seb26** [comment]
I managed to find a solution, sorry seems I came to write an issue too quickly.

I will just read my file with `encoding='utf-8'` param for `open()` and then give Yamale (ultimately the lib I am using to do yaml parsing) the string result via `make_data(content=)`.

The outcome is that I can have emojis/UTF8 characters and still use Resolve API library simultaneously. End of a troublesome few hours.

```
actions_raw_text = open(actions_filepath, 'r', encoding='utf-8').read()
actions_raw = yamale.make_data(content=actions_raw_text)
```
